cdap-standalone/src/main/java/co/cask/cdap/PreviewMain.java

```diff
@@ -0,0 +1,289 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap;
+
+import co.cask.cdap.api.metrics.MetricsCollectionService;
+import co.cask.cdap.app.guice.PreviewServerModule;
+import co.cask.cdap.app.guice.ProgramRunnerRuntimeModule;
+import co.cask.cdap.app.preview.PreviewServer;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.guice.ConfigModule;
+import co.cask.cdap.common.guice.DiscoveryRuntimeModule;
+import co.cask.cdap.common.guice.IOModule;
+import co.cask.cdap.common.guice.LocationRuntimeModule;
+import co.cask.cdap.common.io.URLConnections;
+import co.cask.cdap.common.startup.ConfigurationLogger;
+import co.cask.cdap.common.utils.DirUtils;
+import co.cask.cdap.common.utils.OSDetector;
+import co.cask.cdap.data.runtime.DataFabricModules;
+import co.cask.cdap.data.runtime.DataSetServiceModules;
+import co.cask.cdap.data.runtime.DataSetsModules;
+import co.cask.cdap.data.stream.StreamCoordinatorClient;
+import co.cask.cdap.data2.datafabric.dataset.service.DatasetService;
+import co.cask.cdap.data2.dataset2.DatasetFramework;
+import co.cask.cdap.data2.transaction.stream.StreamAdmin;
+import co.cask.cdap.data2.transaction.stream.StreamConsumerFactory;
+import co.cask.cdap.explore.client.ExploreClient;
+import co.cask.cdap.explore.client.MockExploreClient;
+import co.cask.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import co.cask.cdap.internal.app.runtime.artifact.ArtifactStore;
+import co.cask.cdap.logging.appender.LogAppenderInitializer;
+import co.cask.cdap.logging.guice.LoggingModules;
+import co.cask.cdap.metrics.guice.MetricsClientRuntimeModule;
+import co.cask.cdap.security.authorization.AuthorizerInstantiator;
+import co.cask.cdap.store.guice.NamespaceStoreModule;
+import co.cask.tephra.TransactionManager;
+import co.cask.tephra.inmemory.InMemoryTransactionService;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.Service;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.counters.Limits;
+import org.apache.twill.discovery.InMemoryDiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.List;
+
+/**
+ * Preview Main.
+ */
+public class PreviewMain {
```

Comment:
I think it's better not to have this class now, but rather having the Standalone to start the `PreviewServer` (which it will start anything it needed).
This class should be used in future when we run the preview server in distributed mode, which serves as the entry point for that service (and it should probably name as `PreviewServerTwillRunnable`).